staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -73,6 +75,50 @@ type DRAPlugin interface {
 	internal()
 }
 
+// AllocatedDevice is the devices allocated by dra driver for single ResourceClaim,
+// return by NodePrepareResources func of dra driver.
+type AllocatedDevice struct {
+	// RequestNames is request defined in the ResourceClaimTemplate that this device is associated with.
+	// Optional. If empty, the device is associated with all requests.
+	RequestNames []string
+	// PoolName is the pool that device belongs to.
+	PoolName string
+	// DeviceName is the name of device.
+	DeviceName string
+	// CDIDeviceIDs is  IDs of CDI device associated with device instance. None is also valid.
+	CDIDeviceIDs []string
+}
+
+// DRADriverService defines the interface that should be implemented by DRA driver,
+// different from the previous conventions(1.) that DRA driver implement gRPC
+// interfaces of [drapbv1alpha4.NodeServer] or [drapbv1beta1.DRAPluginServer],
+// instead implement [drapbv1beta1.DRAPluginServer] in this package with DRAPluginServerWrapper,
+// calling functions of DRADriverService interface in DRAPluginServerWrapper.
+type DRADriverService interface {
+	// CheckDeviceAllocation use to prevent from that same device to be assigned to different ResourceClaim.
+	// Demand comes from #128981.
```

Comment:
This number doesn't help a reader of this API documentation. We need write everything down here.